lib/src/options.dart

```diff
@@ -79,6 +80,13 @@ class CommandLineOptions {
   /// source is located.
   final Map<String, String> customUrlMappings;
 
+  /// Whether to use package:dev_compiler for strong static checking.
+  final bool strongMode;
+
+  /// If [strongMode] is enabled, these options configure its behavior,
+  /// otherwise this will be `null`.
+  final StrongModeOptions strongModeOptions;
```

Comment:
On the topic of options, I'm pretty close to landing support for plugins to consume options specified in a shared analysis options file.  Similarly we also have an extension point (almost ready for use) that will allow plugins to read command line options.  With that in place `StrongModeOptions` can stay in the `dev_compiler` proper.

Anyway, just one of many angles to discuss! :)